Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,283 @@
+  simp [hx]
+
+lemma if_eval_neg {r : ℝ} : ∀ᵐ  x : ℝ ∂ volume, (x ∈ {x|x ≥ 0} →
+    ENNReal.ofReal (ite ((x : ℝ) ≥  0) (r * rexp (-(r * x))) 0 ) =
+    ENNReal.ofReal (r * rexp (-(r * x)))) := by
+  apply ae_of_all
+  intro x hx; split_ifs with h; simp only [ge_iff_le] at h
+  · rfl
+  · contrapose h; simp only [ge_iff_le, not_le, not_lt]; exact hx
+
+lemma antiDeriv_tendsto_zero {r : ℝ} (hr : 0 < r) :
+    Tendsto (fun x ↦ -1/r * (exp (-(r * x)))) atTop (𝓝 0) := by
+  rw [← mul_zero (-1/r)]
+  apply Tendsto.mul
+  exact tendsto_const_nhds
+  apply tendsto_exp_neg_atTop_nhds_0.comp
+  exact (tendsto_const_mul_atTop_of_pos hr).2 tendsto_id
+
+open Measure
+
+lemma lintegral_exponentialPdfReal_eq_one (r : ℝ) (hr : 0 < r) :
+    ∫⁻ (x : ℝ), exponentialPdf r x = 1 := by
+  rw [lintegral_split (exponentialPdf r) 0, ←ENNReal.toReal_eq_one_iff];
+  have leftSide : ∫⁻ (x : ℝ) in {x | x < 0}, exponentialPdf r x = 0 := by
+    simp only [exponentialPdf_eq]
+    rw [set_lintegral_congr_fun (isOpen_gt' 0).measurableSet if_eval_pos, lintegral_zero]
+  have rightSide : ∫⁻ (x : ℝ) in {x | x ≥ 0}, exponentialPdf r x
+      = ∫⁻ (x : ℝ) in {x | x ≥ 0}, ENNReal.ofReal (r * rexp (-(r * x))) := by
+    simp only [exponentialPdf_eq]
+    exact set_lintegral_congr_fun isClosed_Ici.measurableSet if_eval_neg
+  rw [leftSide]; simp only [ge_iff_le, add_zero];
+  rw [rightSide, ENNReal.toReal_eq_one_iff, ←ENNReal.toReal_eq_one_iff]
+  have hf : 0 ≤ᵐ[(restrictₗ {x : ℝ | x ≥ 0}) ℙ] (fun x ↦ r * (rexp (-(r * x)))) := by
+    apply ae_of_all
+    intro a
+    positivity
+  rw [← restrictₗ_apply, ← integral_eq_lintegral_of_nonneg_ae hf]
+  · simp only [ge_iff_le, restrictₗ_apply];
+    have IntegrOn : IntegrableOn (fun x ↦ rexp (-(r * x))) (Ioi 0) := by
+      simp only [← neg_mul, exp_neg_integrableOn_Ioi 0 hr]
+    rw [integral_mul_left, Ici_def, integral_Ici_eq_integral_Ioi,
+        integral_Ioi_of_hasDerivAt_of_tendsto' (fun _ _ ↦ antiDeriv_expDeriv_pos' hr) IntegrOn
+        (antiDeriv_tendsto_zero hr)]
+    field_simp
+  · exact ((measurable_id'.const_mul r).neg.exp.const_mul r).stronglyMeasurable.aestronglyMeasurable
+
+/-- The Pdf of the exponential Distribution integrates to 1-/
+@[simp]
+lemma lintegral_exponentialPdf_eq_one (r : ℝ) (hr : 0 < r) :
+    ∫⁻ x, exponentialPdf r x = 1 :=
+  lintegral_exponentialPdfReal_eq_one r hr
+
+end ExponentialPdf
+
+local notation "E" => exponentialPdfReal
+
+open MeasureTheory
+
+/-- Measure defined by the exponential Distribution -/
+noncomputable
+def expMeasure (r : ℝ) : Measure ℝ :=
+  volume.withDensity (exponentialPdf r)
+
+instance instIsProbabilityMeasureExponential (r : ℝ) [Fact (0 < r)] :
+    IsProbabilityMeasure (expMeasure r) where
+  measure_univ := by unfold expMeasure; simp [lintegral_exponentialPdf_eq_one r Fact.out]
+
+section ExponentialCdf
+
+/-- CDF of the exponential Distribution -/
+noncomputable
+def exponentialCdfReal (r : ℝ) : StieltjesFunction :=
+    ProbabilityTheory.cdf (expMeasure r)
```

Comment:
You are in the namespace `ProbabilityTheory`, so you can shorten all the names of that namespace:
```suggestion
    cdf (expMeasure r)
```